backend/src/Parser/CanvasV2.fs

```diff
@@ -0,0 +1,179 @@
+
+  Some(attrName, parseAttrArgs attr.ArgExpr)
+
+/// Update a CanvasModule by parsing a single F# let binding
+/// Depending on the attribute present, this may add a user function, a handler, or a DB
+let parseLetBinding
+  availableTypes
+  (m : CanvasModule)
+  (letBinding : SynBinding)
+  : CanvasModule =
+  match letBinding with
+  | SynBinding (_, _, _, _, attrs, _, _, pat, _returnInfo, expr, _, _, _) ->
+    let expr = ProgramTypes.Expr.fromSynExpr availableTypes expr
+
+    let attrs = attrs |> List.collect (fun l -> l.Attributes)
+
+    let randomIds () : PT.Handler.ids =
+      { moduleID = gid (); nameID = gid (); modifierID = gid () }
+
+    match attrs with
+    | [] ->
+      let newFn = ProgramTypes.UserFunction.fromSynBinding availableTypes letBinding
+      { m with fns = newFn :: m.fns }
+
+    | [ attr ] ->
+      match attr with
+      | SimpleAttribute ("DB", [ name ]) ->
+        // let newDB = PT.DB(name, randomIds ())
+        // { m with dbs = newDB :: m.dbs }
+        Exception.raiseInternal
+          $"Not currently supporting DBs - probably makes sense to wait until they are based on a type rather than 'columns'"
+          [ "attr", attr ]
+
+      | SimpleAttribute ("HttpHandler", [ method; route ]) ->
+        let newHttpHanlder = PT.Handler.Spec.HTTP(route, method, randomIds ())
+        { m with handlers = (newHttpHanlder, expr) :: m.handlers }
+
+      | SimpleAttribute ("REPL", [ name ]) ->
+        //let newHandler = PT.Handler.Spec.REPL(name, randomIds ())
+        //{ m with handlers = (newHandler, expr) :: m.handlers }
+        Exception.raiseInternal
+          $"Not currently supporting REPLs, as we can't test them well yet"
+          [ "attr", attr ]
+
+      | SimpleAttribute ("Worker", [ name ]) ->
+        //let newWorker = PT.Handler.Spec.Worker(name, randomIds ())
+        //{ m with handlers = (newWorker, expr) :: m.handlers }
+        Exception.raiseInternal
+          $"Not currently supporting Workers, as we can't test them well yet"
+          [ "attr", attr ]
+
+      | SimpleAttribute ("Cron", [ name; interval ]) ->
+        //let newCron = PT.Handler.Spec.Cron(name, interval, randomIds ())
+        //{ m with handlers = (newCron, expr) :: m.handlers }
+        Exception.raiseInternal
+          $"Not currently supporting Crons, as we can't test them well yet"
+          [ "attr", attr ]
+
+      | _ ->
+        Exception.raiseInternal
+          $"Not sure how to handle this attribute"
+          [ "attr", attr ]
+
+    | _ ->
+      Exception.raiseInternal
+        $"Can only currently support 1 attribute [<...>] on a let binding"
+        [ "attrs", attrs ]
+
+
+/// An F# module has a list of declarations, which can be:
+/// - a group of let bindings
+/// - a group of type definitions
+/// - a single expression
+/// - a nested module (not currently supported)
+/// - etc
+///
+/// (The declarations are presented as _groups_ presumably to support recursive definitions)
```

Comment:
Not sure what this comment says?

Reply:
Initially I was expecting the "declarations" list of an F# module to be a list of
```
type Declaration =
  | Expr of ..
  | Type of ..
  | Submodule of ..
  | Let of ... 
```

But, to support recursive definitions such as
```
let Enum1 = A of Enum2
and Enum2 = B of Enum1
```
they're presented in groups.

I found this a bit unintuitive so put a note around it. that said, it's probably not worth it, as that's part of the "check the F# compiler docs" note in the README